task_3.py

```diff
@@ -0,0 +1,15 @@
+"""Introduction to programming’: Task 3,
+   Kyshynska Taisiia"""
+import math
+
+print("""Introduction to programming’: Task 3
+         Kyshynska Taisiia""")
+
+x = float(input('x = '))
+z = float(input('z = '))
+y = (x + math.exp(z - 1)) / (1 - (x**2) * abs(x - z))
+
+if (x**2) * abs(x - z) == 1:
```

Comment:
this validation won't save you from the error. 
in case **(x^2) * abs(x - z) == 1** you receive an DivisionByZero error at the line №10